enhancements/network/communication-flows-matrix-ingress.md

```diff
@@ -0,0 +1,245 @@
+  updated and validate it does not change.
+
+### Non-Goals
+
+- Egress traffic.
+
+## Proposal
+
+We introduce a new module to generate an accurate and up-to-date OpenShift 
+communication ingress flows matrix.
+
+- A communication matrix describing the expected flows of incoming traffic will 
+  be included in every OpenShift release documentation.
+
+- A new `oc` command will be added to generate a current snapshot of known 
+  listening ports in a running cluster, `oc adm communication-matrix generate`.
+
+- A new option will be added to OpenShift web console to generate an up-to-date
+  communication matrix.
+
+### Workflow Description
+
+An OpenShift administrator would like to get an accurate and up-to-date OpenShift 
+communication ingress flows matrix.
+
+- The admin reviews OpenShift release documentation to get the included communication
+  matrix describing the expected flows of incoming traffic.
+
+- The admin uses the OpenShift command-line interface (CLI) to generate an up-to-date 
+  communication matrix using the following command `oc adm communication-matrix generate`.
+
+- Optionally, the admin uses the OpenShift web console to generate an up-to-date
+  communication matrix.
+
+### API Extensions
+N/A
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+Out of scope for this proposal.
+
+#### Standalone Clusters
+Out of scope for this proposal.
+
+#### Single-node Deployments or MicroShift
+
+The communication matrix can be generated on single-node deployments.
+
+MicroShift is out of scope for this proposal.
+
+### Implementation Details/Notes/Constraints
+
+A new helper module will be added to `openshift/library-go` to create a communication
+ingress flows matrix from the existing `EndpointSlices` in the OpenShift cluster.
+
+`EndpointSlices` group network endpoints together by unique combinations of 
+protocol, port number, and `Service` name and therefore can be used as the 
+source of truth for cluster traffic. For that to work, every application that 
+attracts ingress traffic running in the cluster should have a `Service` object,
+the control plane automatically creates `EndpointSlices` for any `Service` that 
+has a selector specified. `EndpointSlices` can also be directly created when needed
+or managed by other entities or controllers.
+
+To generate the communication matrix the `EndpointSlices` in the cluster should 
+be filtered by `Service` type `NodePort` and `LoadBalancer` as there are also 
+`EndpointSlices` in the cluster for internal `Services`. The node hosted services 
+such as sshd, rpc, etc. are currently missing `EndpointSlices` (see [Open Questions](#open-questions)).
+
+More info about the Kubernetes `EndpointSlices` resource and management can be found
+in https://kubernetes.io/docs/concepts/services-networking/endpoint-slices/#management
+
+The module allows to generate the communication matrix in various formats:
+- YAML/JSON
+- CSV
+- nftables
```

Comment:
It seems that ANP ingress rules are supported since 4.16: https://docs.openshift.com/container-platform/4.16/rest_api/network_apis/adminnetworkpolicy-policy-networking-k8s-io-v1alpha1.html#spec-ingress
I will add it